spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamController.java

```diff
@@ -145,6 +146,11 @@ public void save(@RequestParam("name") String name,
 			boolean deploy) {
 
 		StreamDefinition stream = new StreamDefinition(name, dsl);
+		if (this.repository.exists(name)) {
```

Comment:
Minor nitpick - seems to make sense to only construct the `StreamDefinition` after checking the repository to see if it exists. I can change this on merge.